packages/legend-query-builder/src/stores/QueryBuilderStateBuilder.ts

```diff
@@ -608,14 +609,26 @@ export const processParameters = (
   queryBuilderState: QueryBuilderState,
 ): void => {
   const queryParameterState = queryBuilderState.parametersState;
-  parameters.forEach((parameter) => {
-    const parameterState = new LambdaParameterState(
-      parameter,
-      queryBuilderState.observableContext,
-      queryBuilderState.graphManagerState.graph,
+  // Here we won't process the parameters which are present in the previous state
+  // because we don't want to lose the parameter value
+  queryParameterState.parameterStates =
+    queryParameterState.parameterStates.filter((p) =>
+      parameters.find((param) => checkIfEquivalent(param, p.parameter)),
     );
-    parameterState.mockParameterValue();
-    queryParameterState.addParameter(parameterState);
+  parameters.forEach((parameter) => {
+    if (
+      queryParameterState.parameterStates.find((ps) =>
+        checkIfEquivalent(ps.parameter, parameter),
+      ) === undefined
+    ) {
+      const parameterState = new LambdaParameterState(
+        parameter,
+        queryBuilderState.observableContext,
+        queryBuilderState.graphManagerState.graph,
+      );
+      parameterState.mockParameterValue();
+      queryParameterState.addParameter(parameterState);
```

Comment:
We could be altering the query with this logic as we could move around how we add the parameter states to querybuilder.parametersState. 
i.e `test: String[1]. name:String[1]` -> (text mode edit) `test2: string[1], name:String[1]`
With this logic i believe we would rewrite the query parameter to `name: String[1], test2: String[1]` when we change querystate to lambda. 
Can't we just for do the `forEach` on each parameter and if we find a similar one i.e `checkIfEquivalent` in the old state we use that one and add that to the parameter state, if not we create a new one and add?